### ISSUE TITLE
💅 `useRegexLiterals` safe fix double-escapes `\\/`, producing an invalid regex literal

### ISSUE THREAD

**Issue by AlbinoGeek** [biomejs/biome#11939]
### Environment information

<details>

```bash
CLI:
  Version:                      2.5.14

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_DISTRIBUTION:           npm
  JS_RUNTIME_NAME:              node
```

</details>

### Rule name

`complexity/useRegexLiterals`

### Playground link

https://biomejs.dev/playground/?lintRules=useRegexLiterals&fix=safeFixes#files.useRegexLiterals.js=YwBvAG4AcwBvAGwAZQAuAGwAbwBnACgAbgBlAHcAIABSAGUAZwBFAHgAcAAoACIAXABcAC8AIgApAC4AdABlAHMAdAAoACIALwAiACkAKQA7AAoA

```js
console.log(new RegExp("\\/").test("/"));
```

The string `"\\/"` holds the two characters `\/`, an escaped slash. The safe fix emits `/\\//`: it escapes the backslash again, so the literal ends after `\\` and the trailing `/.test(...)` no longer parses:

```
SyntaxError: Unexpected token '.'
```

The expected literal is `/\//`. The closed #5693 covers a different case (escapes in template literals).

Reproduced with the 2.5.14 CLI (`biome lint --write`) and in the playground (f74eb38).

### Expected result

An already-escaped `/` in the pattern string is emitted as `\/`, not `\\/`, so the literal matches the same thing as `new RegExp(...)`.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

---

Tool Use Disclaimer: Claude Code using Anthropic Opus 5.5 helped build the reproduction.
